source/slang/slang-check-overload.cpp

```diff
@@ -1438,6 +1438,14 @@ int SemanticsVisitor::CompareLookupResultItems(
             }
         }
 
+        // If we are comparing two generic and one of them in an extension, then this generic
+        // must be CallableDecl. So we're not able to decide which one is better, because candidates
+        // in this pass of overload resolution only check the generic parameters, instead of the
+        // function parameters, therefore, we have to defer this check in the second pass of
+        // overload resolution, where it will check the function parameters.
+        if (as<GenericDecl>(left.declRef.getDecl()))
```

Comment:
I think the case we handled here is probably too narrow.

If *either* left or right are pointing to generic decl itself, we may want to disable more tests, such as `leftIsFreeFormExtension != rightIsFreeFormExtension` above, and the `leftIsGeneric != rightIsGeneric` below.